tests/component/system/test_device_properties.py

```diff
@@ -112,3 +113,91 @@ def test___module___get_chassis___shared_interpreter(device: Device):
     chassis = device.compact_daq_chassis_device
 
     assert chassis._interpreter is device._interpreter
+
+
+def test___ext_cal_last_date_and_time___no_errors(real_x_series_device: Device) -> None:
+    last_date_and_time = real_x_series_device.ext_cal_last_date_and_time
+
+    assert type(last_date_and_time) == datetime
+    assert last_date_and_time.year > 2009
+    assert last_date_and_time.month >= 1
+    assert last_date_and_time.day > 0
+    assert last_date_and_time.hour >= 0
+    assert last_date_and_time.minute >= 0
+
+
+def test___self_cal_last_date_and_time___no_errors(real_x_series_device: Device) -> None:
+    last_date_and_time = real_x_series_device.self_cal_last_date_and_time
+
+    assert type(last_date_and_time) == datetime
+    assert last_date_and_time.year > 2009
+    assert last_date_and_time.month >= 1
+    assert last_date_and_time.day > 0
+    assert last_date_and_time.hour >= 0
+    assert last_date_and_time.minute >= 0
+
+
+def test___device_supports_cal___no_errors(real_x_series_device: Device) -> None:
+    is_cal_supported = real_x_series_device.device_supports_cal
+
+    assert is_cal_supported is True
+
+
+def test___cal_acc_connection_count__raises_attr_not_supported(any_x_series_device: Device) -> None:
+    with pytest.raises(DaqError) as exc_info:
+        _ = any_x_series_device.cal_acc_connection_count
+
+    assert exc_info.value.error_code == DAQmxErrors.ATTR_NOT_SUPPORTED
+
+
+def test___cal_recommended_acc_connection_count_limit___raises_attr_not_supported(
+    any_x_series_device: Device,
+) -> None:
+    with pytest.raises(DaqError) as exc_info:
+        _ = any_x_series_device.cal_recommended_acc_connection_count_limit
+
+    assert exc_info.value.error_code == DAQmxErrors.ATTR_NOT_SUPPORTED
+
+
+def test___cal_user_defined_info___no_errors(real_x_series_device: Device) -> None:
+    user_defined_info = real_x_series_device.cal_user_defined_info
+    info_max_size = real_x_series_device.cal_user_defined_info_max_size
+
+    test_value = "my test value"[:info_max_size]
+    real_x_series_device.cal_user_defined_info = test_value
+    value = real_x_series_device.cal_user_defined_info
+
+    assert info_max_size == len(value)
+    assert value in test_value
+
+    real_x_series_device.cal_user_defined_info = user_defined_info
```

Comment:
This loses the real X Series device's user defined info in error cases. Put a try block around the arrange/act/assert sections and put the cleanup in a finally block.

FYI, this is testing both read and write in the same test, but in this case there's not an obvious way to split them up.

```suggestion
def test___cal_user_defined_info___no_errors(real_x_series_device: Device) -> None:
    try:
        user_defined_info = real_x_series_device.cal_user_defined_info
        info_max_size = real_x_series_device.cal_user_defined_info_max_size

        test_value = "my test value"[:info_max_size]
        real_x_series_device.cal_user_defined_info = test_value
        value = real_x_series_device.cal_user_defined_info

        assert info_max_size == len(value)
        assert value in test_value
    finally:
        real_x_series_device.cal_user_defined_info = user_defined_info
```

Reply:
`try-finally` block is added in the test case.